packages/zowe-explorer/src/Profiles.ts

```diff
@@ -1237,6 +1237,24 @@ export class Profiles extends ProfilesCache {
         }
     }
 
+    public async handleSwitchAuthentication(node?: IZoweNodeType, label?: string): Promise<void> {
```

Comment:
Hey @SanthoshiBoyina, this is a good start for handling this scenario.  We can't assume the profile names. I see the node is passed and not being used, we can use it to get the profile with node.getProfile() which will supply the name of the service profile being used.  Once we have that we can check if it has any secure properties and change appropriately at the service profile level then if not there the base type profile level.  I believe our ZE API has methods in ProfilesCache to get the base profile that is associated with the service profile to obtain the info and name.

Reply:
If label is not being used I don't see a reason to pass to the method.